util/codec/codec.go

```diff
@@ -1209,7 +1209,15 @@ func HashGroupKey(sc *stmtctx.StatementContext, n int, col *chunk.Column, buf []
 				buf[i] = append(buf[i], NilFlag)
 			} else {
 				buf[i] = append(buf[i], decimalFlag)
-				buf[i], err = EncodeDecimal(buf[i], &ds[i], ft.Flen, ft.Decimal)
+				precision := ft.Flen
+				frac := ft.Decimal
+				if precision == 0 {
+					precision, frac = ds[i].PrecisionAndFrac()
+				}
+				if frac > mysql.MaxDecimalScale {
+					frac = mysql.MaxDecimalScale
+				}
```

Comment:
Should we add this to `EncodeDecimal` rather than here? Because `EncodeDecimal` is also called elsewhere, this problem may also occur here?

Reply:
Yes, I think so and I have moved the logic to `EncodeDecimal`.